pkg/api/defaults-custom-cloud-profile.go

```diff
@@ -3,34 +3,113 @@
 
 package api
 
-import "github.com/Azure/aks-engine/pkg/helpers"
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
 
-func (p *Properties) setCustomCloudProfileDefaults() {
+	"github.com/Azure/aks-engine/pkg/helpers"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
 
+func (cs *ContainerService) setCustomCloudProfileDefaults() {
+	p := cs.Properties
 	if p.IsAzureStackCloud() {
+		p.CustomCloudProfile.AuthenticationMethod = helpers.EnsureString(p.CustomCloudProfile.AuthenticationMethod, ClientSecretAuthMethod)
+		p.CustomCloudProfile.IdentitySystem = helpers.EnsureString(p.CustomCloudProfile.IdentitySystem, AzureADIdentitySystem)
+		p.CustomCloudProfile.DependenciesLocation = DependenciesLocation(helpers.EnsureString(string(p.CustomCloudProfile.DependenciesLocation), AzureStackDependenciesLocationPublic))
+		p.SetAzureStackCloudSpec()
+		err := cs.SetCustomCloudProfileEnvironment()
+		if err != nil {
+			log.Fatalf("Failed to set environment - %s", err)
+		}
+	}
+}
 
-		//azureStackCloudSpec is the default configurations for azure stack with public Azure.
-		azureStackCloudSpec := AzureEnvironmentSpecConfig{
-			CloudName: AzureStackCloud,
-			//DockerSpecConfig specify the docker engine download repo
-			DockerSpecConfig: DefaultDockerSpecConfig,
-			//KubernetesSpecConfig is the default kubernetes container image url.
-			KubernetesSpecConfig: DefaultKubernetesSpecConfig,
-			DCOSSpecConfig:       DefaultDCOSSpecConfig,
-			EndpointConfig: AzureEndpointConfig{
-				ResourceManagerVMDNSSuffix: "",
-			},
-			OSImageConfig: map[Distro]AzureOSImageConfig{
-				Ubuntu: DefaultUbuntuImageConfig,
-				RHEL:   DefaultRHELOSImageConfig,
-				CoreOS: DefaultCoreOSImageConfig,
-				AKS:    DefaultAKSOSImageConfig,
-			},
+// SetCustomCloudProfileEnvironment retrives the endpoints from Azure Stack metadata endpoint and set the values for azure.Environment
```

Comment:
Typo: "retrives" -> "retrieves". Also "set" should be "sets," as in `...metadata endpoint and **sets** the values..."